### PR TITLE
pthread: Change pthread_self to function

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

to avoid include unistd.h in pthread.h since link function inside unistd.h is too easy to conflict with the 3rd party library.

## Impact

Minor

## Testing

CI

### CODE REVIEW

File: libs/libc/pthread/pthread_self.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: pthread_self
+ ****************************************************************************/
+
+pthread_t pthread_self(void)
+{
+  return gettid();
```
Comment (pkarashchenko) [inline libs/libc/pthread/pthread_self.c:38]: Do we need a type cast here?

**Review by pkarashchenko — APPROVED** [review]

File: libs/libc/pthread/pthread_self.c (hunk 1 above)
Comment (xiaoxiang781216) [reply libs/libc/pthread/pthread_self.c]: Yes, Done.

**Comment by xiaoxiang781216** [comment]
https://github.com/apache/nuttx/actions/runs/4098214664/jobs/7067171681 is fixed here: https://github.com/apache/mynewt-nimble/pull/1457

**Comment by xiaoxiang781216** [comment]
https://github.com/apache/nuttx/actions/runs/4100097451/jobs/7070641033 is fixed here: https://github.com/spiriou/microADB/pull/34

**Comment by pkarashchenko** [comment]
Somehow the upgrade of microADB didn't solve the CI failure

**Comment by xiaoxiang781216** [comment]
Yes, because nimble has still some issue to fix.

**Comment by yamt** [comment]
> to avoid include unistd.h in pthread.h since link function inside unistd.h is too easy to conflict with the 3rd party library.

what library has a conflicting link()? just curious.

**Pull request closed without merging**